model/src/main/proto/oppia_logger.proto

```diff
@@ -80,6 +110,105 @@ message EventLog {
     int32 sub_topic_id = 2;
   }
 
+  // Structure of a start card context.
+  message StartCardContext {
+    GenericData generic_data = 1;
+    ExplorationData exploration_data= 2;
+    // The linked skill id of the exploration.
+    string skill_id = 3;
+  }
+
+  // Structure of a end card context.
+  message EndCardContext {
+    GenericData generic_data = 1;
+    ExplorationData exploration_data= 2;
+    // The linked skill id of the exploration.
+    string skill_id = 3;
+  }
+
+  // Structure of a hint offered context.
+  message HintOfferedContext {
+    GenericData generic_data = 1;
+    ExplorationData exploration_data= 2;
+    // index of the last revealed hint within the hint list of a state (meaning the index
+    // corresponding to this value is the latest hint that's been revealed by the user)
+    string hint_index = 3;
```

Comment:
Suggest also noting in the documentation what the value of this field is when it's a solution offered. Ditto below for accessed.